packages/material-ui-utils/src/requirePropFactory.js

```diff
@@ -12,14 +12,28 @@ export default function requirePropFactory(componentNameInError) {
   ) => {
     const propFullNameSafe = propFullName || propName;
 
+    // eslint-disable-next-line react/forbid-foreign-prop-types
```

Comment:
I've got this failing test now https://app.circleci.com/pipelines/github/mui-org/material-ui/35659/workflows/8454c729-bc8f-49ba-8ae5-9fde972ca739/jobs/215603

Also, not sure how smart it is to disable this rule - https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-foreign-prop-types.md

I will revert the changes related to this and open a separte PR on this tomorrow.

Reply:
> Also, not sure how smart it is to disable this rule - https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-foreign-prop-types.md

It links https://github.com/oliviertassinari/babel-plugin-transform-react-remove-prop-types to the why. It's fine in our case.

Reply:
Alright, anyway will fix it separately to keep this PR clean